### PR TITLE
Feature/#563 improve dispatch disaggregation

### PR DESCRIPTION
**Pull request by KathiEsterl**
-> solves #563 

-> tested in small networks and on big calculation results

-> left ToDos (low prio) / things to talk about:

- [x] nmp-Constraints -> #621 
- [x] extra_functionalities -> #623
- [x] right now only working with skip_snapshots -> #622 
- [x] slices could be calculated in parallel -> #624 
- [x] too much load_shedding needed?
- [x] add load_shedding to network_tsa when copied although it is set to False, because dispatch_disaggregation is not working otherwise (dependent on fix of #581)

### CODE REVIEW

**Comment by KathiEsterl** [comment]
Concerning the needed load shedding, here is an overview over the results of my test calculations:

4 slices: 2.8 TWh load shedding needed
8 slices: calculation started today
12 slices: 8.3 TWh load shedding needed
24 slices: 16.6 TWh load shedding needed

File: etrago/cluster/snapshot.py [hunk 1]
```diff
@@ -230,218 +255,312 @@ def tsam_cluster(timeseries_df,
...
-                        if i ==-1:
-                            i=0
-                        max_val['SegmentDuration'] = timeseries.index.get_level_values('SegmentDuration')[i]
-                        max_val.set_index(['dates', 'SegmentNo', 'SegmentDuration'],inplace=True)
-                        timeseries.drop(timeseries.index[i], inplace=True)
-                        timeseries = timeseries.append(max_val)
-                        timeseries = timeseries.sort_values(by='dates')
+def segmentation_extreme_periods(
+    timeseries_df, timeseries, extremePeriodMethod
+):
```
Comment (ClaraBuettner) [inline etrago/cluster/snapshot.py:302]: There are no docstrings yet, but you will add them later on on the documentation branch, right?
If yes, it's fine by now.

File: etrago/tools/execute.py [hunk 2]
```diff
@@ -213,12 +242,29 @@ def iterate_lopf(
...
-    if dispatch_disaggregation:
+    if (
+        args["temporal_disaggregation"]["active"] is True
+        and etrago.conduct_dispatch_disaggregation is False
+    ):
+        if not args["csv_export"] is False:
+            path = path + "/temporally_reduced"
+
+        if not args["lpfile"] is False:
+            lp_path = lp_path[0:-3] + "_temporally_reduced.lp"
```
Comment (ClaraBuettner) [inline etrago/tools/execute.py:256]: Is it really needed to store these .lp files? 
It created a lot of additional data

File: etrago/appl.py [hunk 3]
```diff
@@ -521,13 +529,8 @@ def run_etrago(args, json_path):
     # execute etrago function
     print(datetime.datetime.now())
     etrago = run_etrago(args, json_path=None)
+    import pdb
+
+    pdb.set_trace()
```
[the remark this answers is not in the document]
Comment (KathiEsterl) [reply etrago/appl.py:534]: Deleted

File: etrago/cluster/snapshot.py (hunk 1 above)
Comment (KathiEsterl) [reply etrago/cluster/snapshot.py:302]: Yes, they are added in the documentation branch: https://github.com/openego/eTraGo/tree/documentation-release0.9.0

File: etrago/appl.py [hunk 4]
```diff
@@ -134,19 +134,22 @@
...
-        "n_segments": 5,
-    },  # number of segments - only relevant for segmentation
+        "n_segments": 5,  # number of segments - only relevant for segmentation
+    },
     "skip_snapshots": 5,  # False or number of snapshots to skip
-    "dispatch_disaggregation": False,  # choose if full complex dispatch optimization should be conducted
+    "temporal_disaggregation": {
+        "active": True,  # choose if temporally full complex dispatch optimization should be conducted
+        "no_slices": 8,  # number of subproblems optimization is divided into
+    },
```
[the remark this answers is not in the document]
Comment (KathiEsterl) [reply etrago/appl.py:152]: Done

**Review by ClaraBuettner — APPROVED** [review]

**Pull request merged**